src/charm.py

```diff
@@ -115,6 +115,7 @@ def generate_and_save_config(self):
 
         nginx_template = env.get_template("src/nginx.conf.template")
         nginx_template = nginx_template.stream(
+            controller_ws_api=data.get("controller-url").replace("wss", "https"),
```

Comment:
Can you add a comment here as to why this is necessary. IMHO it's kind of confusing.